test/unit/server/get_ledger_entries_test.js

```diff
@@ -0,0 +1,262 @@
+        jsonrpc: '2.0',
+        id: 1,
+        method: 'getLedgerEntries',
+        params: [
+          [ledgerExpirationKey.toXDR('base64'), ledgerKey.toXDR('base64')]
+        ]
+      })
+      .returns(
+        Promise.resolve({
+          data: {
+            result: {
+              latestLedger: 420,
+              entries: [
+                {
+                  lastModifiedLedgerSeq: expectedResult.lastModifiedLedgerSeq,
+                  key: ledgerExpirationKey.toXDR('base64'),
+                  xdr: ledgerExpirationEntry.toXDR('base64')
+                },
+                {
+                  lastModifiedLedgerSeq: expectedResult.lastModifiedLedgerSeq,
+                  key: ledgerKey.toXDR('base64'),
+                  xdr: ledgerEntry.toXDR('base64')
+                }
+              ]
+            }
+          }
+        })
+      );
+
+    this.server
+      .getLedgerEntries(ledgerKey)
+      .then((response) => {
+        expect(response.entries).to.have.lengthOf(1);
+        let result = response.entries[0];
+        expect(result.key.toXDR('base64')).to.be.deep.equal(
+          expectedResult.key.toXDR('base64')
+        );
+        expect(result.val.toXDR('base64')).to.be.deep.equal(
+          expectedResult.val.toXDR('base64')
+        );
+        expect(result.expiration.toXDR('base64')).to.be.deep.equal(
+          expectedResult.expiration.toXDR('base64')
+        );
+        done();
+      })
+      .catch((err) => done(err));
+  });
+
+  it('ledger expiration key is requested by caller, no expiration meta needed on response', function (done) {
+    let expectedResult = {
+      lastModifiedLedgerSeq: 1,
+      key: ledgerExpirationKey,
+      val: ledgerExpirationEntry
+    };
+
+    this.axiosMock
+      .expects('post')
+      .withArgs(serverUrl, {
+        jsonrpc: '2.0',
+        id: 1,
+        method: 'getLedgerEntries',
+        params: [[ledgerExpirationKey.toXDR('base64')]]
+      })
+      .returns(
+        Promise.resolve({
+          data: {
+            result: {
+              latestLedger: 420,
+              entries: [
+                {
+                  lastModifiedLedgerSeq: expectedResult.lastModifiedLedgerSeq,
+                  key: ledgerExpirationKey.toXDR('base64'),
+                  xdr: ledgerExpirationEntry.toXDR('base64')
+                }
+              ]
+            }
+          }
+        })
+      );
+
+    this.server
+      .getLedgerEntries(ledgerExpirationKey)
+      .then((response) => {
+        expect(response.entries).to.have.lengthOf(1);
+        let result = response.entries[0];
+        expect(result.key.toXDR('base64')).to.be.deep.equal(
+          expectedResult.key.toXDR('base64')
+        );
+        expect(result.val.toXDR('base64')).to.be.deep.equal(
+          expectedResult.val.toXDR('base64')
+        );
+        expect(result.expiration).to.be.undefined;
+        done();
+      })
+      .catch((err) => done(err));
+  });
+});
```

Comment:
it looks like the assertions are identical in each of these cases. there's a lot of verbosity in the setup; could we combine the setup code into a single blob and do a loop over the expected/actual scenarios?

great coverage, though!

Reply:
fair enough!